### PR TITLE
Improve the Draggable component

### PR DESCRIPTION
**Pull request by Alyar666**

### CODE REVIEW

File: js/ui/draggable.js [hunk 1]
```diff
@@ -42,85 +43,145 @@ var Draggable = DOMComponent.inherit({
...
+            position.top = e.pageY - $element.offset().top + translator.locate($element).top - $element.height() / 2;
         }
 
-        this._move(position);
+        this._move(position, $element);
 
         this._getAction("onDrag")({ event: e });
     },
 
+    _isValidateElement: function(event, $element) {
```
Comment (vconst) [inline js/ui/draggable.js]: _isValidateElement => _isValidElement | _validateElement

File: js/ui/draggable.js [hunk 2]
```diff
@@ -42,85 +43,145 @@ var Draggable = DOMComponent.inherit({
...
+        let handle = this.option("handle"),
+            isHandleElement = handle && $(event.originalEvent.target).is(handle);
```
Comment (vconst) [inline js/ui/draggable.js]: is -> closest?

File: styles/widgets/common/draggable.less [hunk 3]
```diff
@@ -8,3 +8,15 @@
         cursor: default;
     }
 }
+
+.dx-draggable-dragging {
```
Comment (vconst) [inline styles/widgets/common/draggable.less]: Check this class

File: testing/tests/DevExpress.ui.widgets/draggable.tests.js [hunk 4]
```diff
@@ -429,3 +468,333 @@ QUnit.test("element position on click should be updated considering direction",
...
+    this.createDraggable({
+        dragTemplate: function() {
+            return $("<div id='myDragElement'/>").text("test");
+        }
+    });
+
+    this.pointer.down().move(10, 10);
+
+    // assert
+    assert.strictEqual($("body").children("#myDragElement").length, 1, "there is a cloned element");
```
Comment (vconst) [inline testing/tests/DevExpress.ui.widgets/draggable.tests.js:692]: check that dx-draggable-clone class is added

File: testing/tests/DevExpress.ui.widgets/draggable.tests.js [hunk 5]
```diff
@@ -429,3 +468,333 @@ QUnit.test("element position on click should be updated considering direction",
...
+    // arrange
+    this.createDraggable({
+        handle: "#handle"
+    });
+
+    // act
+    this.pointer.down().move(10, 10).up();
+
+    // assert
+    this.checkPosition(0, 50, assert);
```
Comment (vconst) [inline testing/tests/DevExpress.ui.widgets/draggable.tests.js:794]: add description

File: testing/tests/DevExpress.ui.widgets/draggable.tests.js [hunk 6]
```diff
@@ -429,3 +468,333 @@ QUnit.test("element position on click should be updated considering direction",
...
+
+    // act
+    pointerMock(this.$element.find("#handle").first()).start().down().move(10, 10);
+
+    // assert
+    this.checkPosition(10, 60, assert);
```
Comment (vconst) [inline testing/tests/DevExpress.ui.widgets/draggable.tests.js:800]: add description

**Review by vconst — CHANGES_REQUESTED** [review]

**Review by vconst — APPROVED** [review]

**Review by jtoming830 — APPROVED** [review]

**Pull request merged**